src/Confluent.SchemaRegistry.Serdes.Avro/GenericSerializerImpl.cs

```diff
@@ -131,14 +134,26 @@ public async Task<byte[]> Serialize(string topic, GenericRecord data, bool isKey
                     if (!registeredSchemas.Contains(subjectSchemaPair))
                     {
                         int newSchemaId;
-                        // first usage: register/get schema to check compatibility
-                        if (autoRegisterSchema)
+                        if (useLatestVersion)
                         {
-                            newSchemaId = await schemaRegistryClient.RegisterSchemaAsync(subject, writerSchemaString).ConfigureAwait(continueOnCapturedContext: false);
+                            var latestSchema = await schemaRegistryClient.GetLatestSchemaAsync(subject)
+                                .ConfigureAwait(continueOnCapturedContext: false);
+                            newSchemaId = latestSchema.Id;
```

Comment:
the latest schema is thrown away - isn't being used to serialize the data, so aren't there scenarios in which this breaks? (if schema changes is not forward compatible)?

Reply:
Yes, `use.latest.version=true` is an advanced option.  For the Java client, we can optionally check that the latest schema is backward compatible with the schema of the object being passed.   However, we don't have the code for compatibility checks available in C#.

Reply:
I think it's a bit dangerous. Would definitely note this in the property docstrings. maybe "WARNING: There is no check that the latest schema is backwards compatible with the schema used for serialization". Also, I like the idea of changing the config name to `dangerously.use.latest.version`.

Reply:
I'll add the docstring but would like to keep the config name the same as the Java config name.

Reply:
ok. i do have a definite preference for having the prefix, but not strong enough to push the point.

Reply:
can you also update the Python PR? thanks!

Reply:
Yup, updated the Python PR as well